core/primitives/src/state_record.rs

```diff
@@ -28,3 +39,101 @@ pub enum StateRecord {
         data: Option<Vec<u8>>,
     },
 }
+
+impl StateRecord {
```

Comment:
It is not a part of this PR, but I want to highlight @evgenykuzyakov and @frol that we should not be storing data inside `StateRecord` as base64 string. Instead it should be `Vec<u8>` and we should use serde helpers to serialize it in base64 string.

Reply:
Yes, it needs some work.

Reply:
Yes, I noticed that when I was doing those configs moving, but had never had a chance to work on fixing this.